### ISSUE TITLE
ocaml `let switch = 3` prints as `let match = 3;`

### ISSUE THREAD

**Issue by chenglou** [facebook/reason#1093]

**Comment by SanderSpies** [comment]
Converting OCaml to Reason? This is intended.

**Comment by chenglou** [comment]
lmao, cc @bobzhang. I guess I kinda get it? Leaving this open til it's documented

**Comment by SanderSpies** [comment]
https://github.com/facebook/reason/blob/master/src/syntax_util.ml#L133

**Comment by SanderSpies** [comment]
To clarify a bit more: within Reason `let switch` is a syntax error while in OCaml `let match` is a syntax error.

**Comment by cristianoc** [comment]
Swapping strings seems limited in this case.
Do we have infra to substitute this way?

1) ml -> re
`match` -> `switch`
`switch` -> `switch_`
`match_` -> `match`
`switch_` -> `switch__`
`match__` -> `match_`
`switch__` -> `switch___`
`match___` -> `match__`
...

2) re -> ml
`switch` -> `match`
`match` -> `match_`
`switch_` -> `switch`
`match_` -> `match__`
`switch__` -> `switch_`
`match__` -> `match___`
`switch___` -> `switch__`

So e.g.
`match switch with`
     ml -> re
`switch switch_ {`
    re -> ml
`match switch with`

**Comment by bobzhang** [comment]
name mangling is something I would like to avoid if we can.

```re
switch x {
  case None => 0;
  case Some x => x 
}

```

```re
match x {
  case None => 0;
  case Some x => x
}
```
They are mostly the same?

**Comment by jordwalke** [comment]
@cristianoc I really like your suggestion - it's simple, we should definitely do it.

**Comment by yunxing** [comment]
@cristianoc We don't have the infra to substitute this way but I think we should do this.

**Comment by chenglou** [comment]
Fixed through #1539
